core/unix/signal.c

```diff
@@ -2676,12 +2675,16 @@ thread_set_self_context(void *cxt)
     memset(&frame, 0, sizeof(frame));
 #ifdef LINUX
 #    ifdef X86
+    dcontext_t *dcontext = get_thread_private_dcontext();
```

Comment:
It seems fragile to put this decl in a different ifdef set from the save_fpstate usage below.

Reply:
Separated the declaration into its own ifdef.